app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinFragment.java

```diff
@@ -184,4 +238,41 @@ private void createNewPin() {
             });
         });
     }
+
+    private String pinniesToString(Long pinniesCount) {
+        String tempString;
+
+        if (pinniesCount < 999) {
+            return pinniesCount.toString();
+        } else if (pinniesCount < 999999) {
+            tempString = pinniesCount/1000 + "." + (pinniesCount % 1000) / 100 + "" + (pinniesCount % 100) / 10 + "K";
+        } else if (pinniesCount < 999999999) {
+            tempString = pinniesCount / 1000000 + "." + (pinniesCount % 1000000) / 100000 + "" + (pinniesCount % 100000) / 10000 + "M";
+        }
+        else return pinniesCount.toString();
+
+        if(tempString.charAt(3) == '.'){
+            return tempString.substring(0, 3) + tempString.charAt(tempString.length() - 1);
+        }
+        else{
+            return tempString.substring(0, 4) + tempString.charAt(tempString.length() - 1);
+        }
+    }
+
+    private void setPinniesUI(){
+        // I'm so sorry
+        if (pinnieCount == null || pinCost == null) return;
+
+        if (pinnieCount >= pinCost){
+            dropButton.setEnabled(true);
+        } else {
+            insufficientPinniesText.setVisibility(getView().VISIBLE);
+        }
+
+        topBarText.setText("Balance: " + pinniesToString(pinnieCount));
+        dropButton.setText("Drop Pin - " + pinniesToString(pinCost));
```

Comment:
Can we extract these strings with a placeholder and then call `String.format` to replace the placeholder with our desired value?

Reply:
Looks like they already existed in the strings.xml, so I replaced with those references.